Implement a polling version of watch callback in CommandIsolator

St0rmingBr4in opened the pull request.

Commit:
Implement a polling version of watch callback in CommandIsolator

Signed-off-by: Julien DOCHE <j.doche@criteo.com>

kamaradclimber commented on src/CommandIsolator.cpp:
```diff
@@ -117,27 +123,38 @@ process::Future<ContainerLimitation> CommandIsolatorProcess::watch(
+                                 .run(command, inputStringified);
+        return output;
+      },
+      [this, command](Try<string> output) -> ControlFlow<ContainerLimitation> {
```
pass minimum info (this->m_debug)

kamaradclimber commented on src/Command.hpp:
```diff
@@ -17,22 +18,33 @@ const unsigned long DEFAULT_COMMAND_TIMEOUT = 30;
-      : m_cmd(command), m_timeout(DEFAULT_COMMAND_TIMEOUT) {}
+      : m_cmd(command),
+        m_timeout(DEFAULT_COMMAND_TIMEOUT),
+        m_frequence(DEFAULT_COMMAND_FREQUENCE) {}
```
about frequency: I'm a bit sad that we need to add a parameter for all commands, but used only for command used in `usage`. Two options:
- inherit from Command class and add frequency in subclass
- don't allow user to configure frequency and compute it based on other parameters (timeout for instance)

Commit:
Add recurrent command definition

Signed-off-by: Julien DOCHE <j.doche@criteo.com>

kamaradclimber commented on src/Command.hpp:
```diff
@@ -35,6 +36,33 @@ class Command {
 
+class RecurrentCommand : public Command {
+ public:
+  explicit RecurrentCommand(const Command& command)
```
it it necessary to have all those constructors? I don't think we need them all and might decide to provide only one

Commit:
Remove unused constructors

Signed-off-by: Julien DOCHE <j.doche@criteo.com>

Commit:
Pass minimum context to lambda

Signed-off-by: Julien DOCHE <j.doche@criteo.com>

Commit:
Terminate the loop when the Future has a discard

This is not the best way of stopping the loop. The ideal way would be that
libprocess stops the loop whenever the future associated receives a discard.
Unfortunately this is not yet the case.
